app/util/appUtil.js

```diff
@@ -125,6 +125,34 @@ async function getLastTokenId() {
   return lastTokenId ? parseInt(lastTokenId, 10) : 0
 }
 
+async function containsInvalidMembershipOwners(outputs) {
+  const membershipMembers = await getMembers(false)
+
+  const validOwners = outputs.reduce((acc, { owner }) => {
+    if (membershipMembers.includes(owner)) {
+      acc.push(owner)
+      return acc
+    }
+  }, [])
+
+  return !validOwners || validOwners.length === 0 || validOwners.length !== outputs.length
+}
+
+function membershipReducer(members) {
+  return members.reduce((acc, item) => {
+    acc.push({ address: item })
+    return acc
+  }, [])
+}
+
+async function getMembers(reduceMembers = true) {
```

Comment:
This is a little ugly form an API point of view as the boolean completely changes the type structure. Maybe consider not executing the reducer inside this method at all (just return `result`) and then exporting and calling the `membershipReducer` from the route?